ui/src/router.ts

```diff
@@ -0,0 +1,59 @@
+import * as svelteStore from "svelte/store";
+
+export type ProfileTab = "projects" | "following" | "funding";
+export type ProjectTab = "files" | "commits" | "commit";
+export type UserProfileTab = "projects";
+
+export type Route =
+  | { type: "empty" }
+  | { type: "designSystemGuide" }
+  | { type: "lock" }
+  | { type: "onboarding" }
+  | { type: "profile"; activeTab: ProfileTab }
+  | { type: "userProfile"; activeTab: UserProfileTab; urn: string }
+  | {
+      type: "project";
+      activeTab: ProjectTab;
+      urn: string;
+      commitHash: string | null;
+    }
+  | { type: "settings" };
+
+const persistedState: Route | null = window.history.state;
```

Comment:
I believe the History API persists its state only within the current session. If I'd stop the app and start it again, it would start with an empty history. I don't see a problem with how it is at the moment.